Detectors/EMCAL/calibration/include/EMCALCalibration/EMCALChannelCalibrator.h

```diff
@@ -0,0 +1,128 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \class EMCALChannelCalibrator
+/// \brief  Perform the EMCAL bad channel calibration
+/// \author Hannah Bossi, Yale University
+/// \ingroup EMCALCalib
+/// \since Feb 11, 2021
+
+#ifndef EMCAL_CHANNEL_CALIBRATOR_H_
+#define EMCAL_CHANNEL_CALIBRATOR_H_
+
+#include "DetectorsCalibration/TimeSlotCalibration.h"
+#include "DetectorsCalibration/TimeSlot.h"
+#include "DataFormatsEMCAL/Cell.h"
+#include "EMCALBase/Geometry.h"
+#include "CCDB/CcdbObjectInfo.h"
+
+#include <array>
+#include <boost/histogram.hpp>
+
+namespace o2
+{
+namespace emcal
+{
+class EMCALChannelData
+{
+  //using Slot = o2::calibration::TimeSlot<o2::emcal::EMCALChannelData>;
+  using Cells = o2::emcal::Cell;
+  using boostHisto = boost::histogram::histogram<std::tuple<boost::histogram::axis::regular<double, boost::use_default, boost::use_default, boost::use_default>, boost::histogram::axis::integer<>>, boost::histogram::unlimited_storage<std::allocator<char>>>;
+
+ public:
+  // NCELLS includes DCal, treat as one calibration
+  o2::emcal::Geometry* mGeometry = o2::emcal::Geometry::GetInstanceFromRunNumber(300000);
+  int NCELLS = mGeometry->GetNCells();
+
+  EMCALChannelData(int nb, float r) : mNBins(nb), mRange(r)
+  {
+    // boost histogram with amplitude vs. cell ID, specify the range and binning of the amplitude axis
+    mHisto = boost::histogram::make_histogram(boost::histogram::axis::regular<>(mNBins, 0, mRange, "t-texp"), boost::histogram::axis::integer<>(0, NCELLS, "CELL ID"));
+  }
+
+  ~EMCALChannelData() = default;
+
+  /// \brief Print relevant info for EMCALChannelData on a given stream
+  /// \param stream Stream on which the info is printed on
+  /// The function is called in the operator<< providing direct access
+  /// to protected members. Explicit calls by the users is normally not
+  /// necessary.
+  void PrintStream(std::ostream& stream) const;
+  /// \brief Print a useful message about the container.
+  void print();
+  /// \brief Fill the container with the cell ID and amplitude.
+  void fill(const gsl::span<const o2::emcal::Cell> data);
+  /// \brief Merge the data of two slots.
+  void merge(const EMCALChannelData* prev);
```

Comment:
Not sure how this is handled in the interface as this is a templated object, but if possible I would move this to reference and ban pointers as much as possible.